src/coreclr/jit/loopcloning.cpp

```diff
@@ -1107,15 +1148,22 @@ bool Compiler::optDeriveLoopCloningConditions(unsigned loopNum, LoopCloneContext
             return false;
         }
 
-        // GT_LT loop test: limit <= arrLen
-        // GT_LE loop test: limit < arrLen
+        // Increasing loops
+        // GT_LT loop test: (start < end) ==> (end <= start)
+        // GT_LE loop test: (start <= end) ==> (end < start)
+        //
+        // Decreasing loops
+        // GT_GT loop test: (end > start) ==> (end <= start)
+        // GT_GE loop test: (end >= start) ==> (end < start)
```

Comment:
These conditions don't make sense.

Shouldn't it be, e.g.:
```suggestion
        // GT_LT loop test: (start < end) ==> (end <= arrLen)
```

Reply:
yes, I too was confused initially but forgot to revisit this.